### ISSUE TITLE
flutter plugin publish error

### ISSUE THREAD

**Issue by espanic** [dart-lang/pub#3422]
flutter packages pub publish 

Then, this weird error appears. 

Invalid argument(s): Only absolute normalized paths are supported: ..
package:analyzer/src/dart/analysis/analysis_context_collection.dart 121:7  AnalysisContextCollectionImpl._throwIfNotAbsoluteNormalizedPa
th
package:analyzer/src/dart/analysis/analysis_context_collection.dart 112:7  AnalysisContextCollectionImpl._throwIfAnyNotAbsoluteNormalize
dPath
package:analyzer/src/dart/analysis/analysis_context_collection.dart 49:5   new AnalysisContextCollectionImpl
package:pub/src/null_safety_analysis.dart 178:35                           NullSafetyAnalysis.nullSafetyComplianceOfPackages.<fn>
===== asynchronous gap ===========================
package:pub/src/null_safety_analysis.dart 145:11                           NullSafetyAnalysis.nullSafetyComplianceOfPackages
===== asynchronous gap ===========================
package:pub/src/validator/null_safety_mixed_mode.dart 26:28                NullSafetyMixedModeValidator.validate
===== asynchronous gap ===========================
package:pub/src/command/lish.dart 239:5                                    LishCommand._validate
===== asynchronous gap ===========================
package:pub/src/command/lish.dart 213:9                                    LishCommand.runProtected
This is an unexpected error. Please run

    dart pub --trace publish

and include the logs in an issue on https://github.com/dart-lang/pub/issues/new
pub finished with exit code 1

Any solution to fix this error?

**Comment by sigurdm** [comment]
What is the output of `flutter version`?
Can you reveal the `pubspec.yaml`.

**Comment by shengrun-hub** [comment]
Have you solved this problem？

**Comment by sigurdm** [comment]
This looks like a bug in how we integrate with the analyzer.

@espanic and @shengrun-hub  do any of you have any more context? A reproduction would help immensely in tracking down this.

**Comment by gredajustyna** [comment]
> This looks like a bug in how we integrate with the analyzer.
> 
> @espanic and @shengrun-hub do any of you have any more context? A reproduction would help immensely in tracking down this.

I came across a similar issue today. Not sure if my context is the same one as theirs, but it happened when i tried to publish a plugin which referenced my own plugin, not published to `pub.dev`. I imported the folder of my own plugin to the root folder of the main plugin and added it in `pubspec.yaml`:

```yaml=
sodium_key_generator_plugin:
    path: sodium_key_generator_plugin
```

And the path is exatly what causes the error. I am aware this may not be a good practice, but as such issue occured, I'd be glad to have some explanation for it.
